lib/game.rb

```diff
@@ -0,0 +1,85 @@
+class Game
+          WIN_COMBINATIONS =
+          [[0,1,2],
+          [3,4,5],
+          [6,7,8],
+          [0,3,6],
+          [1,4,7],
+          [2,5,8],
+          [0,4,8],
+          [6,4,2]]
+
+
+          attr_accessor :board, :player_1, :player_2
+
+    def initialize(player_1=Player::Human.new("X"),player_2=Player::Human.new("O"),board=Board.new)
+      @player_1 = player_1
+      @player_2 = player_2
+      @board = board
+    end
+
+    def current_player
+      if board.turn_count % 2 == 0
+        player_1
+      else
+        player_2
+      end
+    end
+
+    def over?
+      won? || draw?
+    end
+
+    def won?
+      WIN_COMBINATIONS.each do |check|
+      win_index_1 = check[0]
+      win_index_2 = check[1]
+      win_index_3 = check[2]
+
+      position_1 = board.cells[win_index_1]
+      position_2 = board.cells[win_index_2]
+      position_3 = board.cells[win_index_3]
+
+      if position_1 == "X" && position_2 == "X" && position_3 == "X" || position_1 == "O" && position_2 == "O" && position_3 == "O"
+        return true
+      end
+      end
+        return false
+    end
+
+    def draw?
+      !won? && board.full?
+    end
+
+    def winner
```

Comment:
```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.